test-framework/keycloak-server/src/main/java/io/quarkus/test/keycloak/server/KeycloakTestResourceLifecycleManager.java

```diff
@@ -51,13 +62,19 @@
         } else {
             throw new ConfigurationException("Please set either 'keycloak.docker.image' or 'keycloak.version' system property");
         }
+
         keycloak = new GenericContainer<>(keycloakDockerImage)
                 .withExposedPorts(8080, 8443)
                 .withEnv("DB_VENDOR", "H2")
                 .withEnv("KEYCLOAK_USER", "admin")
                 .withEnv("KEYCLOAK_PASSWORD", "admin")
                 .waitingFor(Wait.forHttp("/auth").forPort(8080));
 
+        if (KEYCLOAK_USE_HTTPS && KEYCLOAK_TRUSTSTORE_REQUIRED) {
+            keycloak = keycloak.withClasspathResourceMapping(KEYCLOAK_TLS_KEY, KEYCLOAK_TLS_KEY_MOUNTED_PATH,
```

Comment:
@pedroigor is it the right way to mount a key ? I've read in in a rather old Stackoverflow that this is how it should be done, but I'm not sure it is still the case, the test does not work

Reply:
@pedroigor I have not seen in the latest docs any references to a private key having to be mounted to `/etc/x509/http/tls.key`, it is only mentioned in the non up to date docs. I wonder what is the current approach is - mounting a private key should lead to WildFly converting it to a keystore, but perhaps now an actual keystore file should be mapped ?

Reply:
Perhaps it should be `/etc/x509/https/tls.key`?

Reply:
@pedroigor thanks, the docs at https://hub.docker.com/r/jboss/keycloak/ do refer to `/etc/x509/https`.

I'm pretty sure some progress has been made as a result since now I'm seeing now in the logs:

```
Creating HTTPS keystore via OpenShift's service serving x509 certificate secrets..
HTTPS keystore not created at: /opt/jboss/keycloak/standalone/configuration/keystores/https-keystore.jks (check permissions?)
```

I've checked, this error is not logged if the path is wrong (`http` in `/etc/x509/http/tls.key`).

The section at https://hub.docker.com/r/jboss/keycloak/ points out at the possible problem.

The permissions for both `tls.crt` and `tls.key` are `r` for the world on my system.
Host mounting them with `withClasspathResourceMapping` or copying to the docker (`.withCopyFileToContainer`) makes no difference though.... 
But it is nearly there, we just need to see how it is done in some working deployments

Reply:
@pedroigor @rvansa the example [here](https://stackoverflow.com/questions/49859066/keycloak-docker-https-required) shows how these files are mapped with the specific user/group ids and also the mode. `TestContainers` does now allow to set it when mounting the files.
I've also seen somewhere setting the mode to `655` on the local system before mounting the files should work - but I could not make it work.